pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -211,21 +214,22 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 	}
 
 	glog.V(3).Infof("eviction manager: synchronize housekeeping")
-
 	// build the ranking functions (if not yet known)
 	// TODO: have a function in cadvisor that lets us know if global housekeeping has completed
-	if len(m.resourceToRankFunc) == 0 || len(m.resourceToNodeReclaimFuncs) == 0 {
-		// this may error if cadvisor has yet to complete housekeeping, so we will just try again in next pass.
-		hasDedicatedImageFs, err := diskInfoProvider.HasDedicatedImageFs()
-		if err != nil {
+	if m.dedicatedImageFs == nil {
```

Comment:
I think we still want to enter this if `len(m.resourceToRankFunc) == 0 || len(m.resourceToNodeReclaimFuncs) == 0`.
So this should be `if m.dedicatedImageFs == nil || len(m.resourceToRankFunc) == 0 || len(m.resourceToNodeReclaimFuncs) == 0`